0014-http-ilp/0014-http-ilp.md

```diff
@@ -0,0 +1,249 @@
+
+### 1.2. Server admin sets up a new receiver
+
+Ankita uses the web-based UI of her ILP wallet provider to create a new receiver tied to her account. She chooses an arbitrary unique (for her account) `name` to identify this receiver and enters the `webhook` URI from the previous step which points to her `myservice.example` server. Internally, the UI makes a call to the wallet provider's backend:
+
+###### Request
+
+``` http
+POST /api/receivers
+Host: nexus.justmoon.com
+Content-Type: application/json
+```
+``` js
+{
+  "name": "filepay",
+  "webhook": "https://myservice.example/webhook"
+}
+```
+
+###### Response
+
+``` js
+{
+  "name": "filepay",
+  "webhook": "https://myservice.example/webhook",
+  "secret": "fxPERNaS4FGlC8H7eg6UfYVlglmFynFc8nh5la9PBGM"
+}
+```
+
+Afterwards, the UI shows Ankita the receiver address and secret:
+
+* SPSP Receiver Address: `ankita+filepay@nexus.justmoon.com`
+* SPSP Receiver Secret: `fxPERNaS4FGlC8H7eg6UfYVlglmFynFc8nh5la9PBGM`
+
+This secret will henceforth be referred to as the *receiver secret* / `receiver_secret`.
+
+
+### 1.3. Server admin enters credentials into the HTTP-ILP server module config
+
+Next, Ankita logs back into her server and edits a config file of the HTTP-ILP server module to enter the receiver address and secret she obtained:
+
+``` ini
+[spsp_creds]
+receiver_address=ankita+filepay@nexus.justmoon.com
+receiver_secret=fxPERNaS4FGlC8H7eg6UfYVlglmFynFc8nh5la9PBGM
+```
+
+Next, she restarts her server to load the new configuration.
+
+### 1.4. Paid HTTP server fetches receiver information
+
+When the HTTP-ILP server module loads up, it uses [Webfinger](https://tools.ietf.org/html/rfc7033) to look up the `receiver_address` from the configuration.
+
+###### Request
+``` http
+GET /.well-known/webfinger?uri=ankita+filepay@nexus.justmoon.com
+Host: nexus@justmoon.com
+```
+
+The wallet will generate an ILP address using its ledger prefix, the fixed term `~recv`, the username `ankita` and the receiver name `filepay`.
+
+* ILP Address: `us.nexus.ankita.~recv.filepay`
+
+Note that the method for mapping an SPSP address to an ILP address is totally arbitrary and not part of this standard. The wallet MAY do this any way that it wants. In practice, the wallet may choose an address not including the username for privacy reasons.
+
+###### Response
+``` http
+HTTP/1.1 200 OK
+Content-Type: application/json
+```
+``` js
+{
+  // TODO ...
+}
+```
+
+The HTTP-ILP server module now knows the ILP address and caches it in memory for a certain period (depending on the cache header of the above request.) The reason we had Ankita enter the SPSP address and not the ILP address is because the ILP address may change over time as ILP providers change their routing structure. Note that ILP may support multihoming (this is still being debated as of this writing), so there may also be more than one ILP address.
```

Comment:
> Note that ILP may support multihoming (this is still being debated as of this writing), so there may also be more than one ILP address.

This can be removed